### PR TITLE
injections(lua): highlight Lua code in vim.rpcrequest and vim.rpcnotify

### PR DESCRIPTION
**Pull request by smjonas**
Before:
image

After:
image

### CODE REVIEW

File: queries/lua/injections.scm [hunk 1]
```diff
@@ -16,7 +16,14 @@
   arguments: (arguments (string content: _ @query) .))
   (#any-of? @_vimcmd_identifier "vim.treesitter.query.set" "vim.treesitter.query.parse_query" "vim.treesitter.query.parse"))
 
-;; highlight string as query if starts with `;; query`
+; vim.rcprequest(123, "nvim_exec_lua", "return vim.api.nvim_buf_get_lines(0, 0, -1, false)", false)
+((function_call
+  name: (_) @_vimcmd_identifier
+  arguments: (arguments . (_) . (string) @_method . (string content: _ @lua)))
+  (#any-of? @_vimcmd_identifier "vim.rpcrequest" "vim.rpcnotify")
+  (#eq? @_method "\"nvim_exec_lua\""))
```
Comment (lucario387) [inline queries/lua/injections.scm:24]: hmm, is there any other methods that also needs lua highlighting?

**Review by lucario387 — APPROVED** [review]

File: queries/lua/injections.scm (hunk 1 above)
Comment (clason) [reply queries/lua/injections.scm:24]: There's a number of new functions in 0.9.0; see the upstream queries.

Comment (lucario387) [reply queries/lua/injections.scm:24]: > There's a number of new functions in 0.9.0; see the upstream queries.

From the upstream lists, nvim-treesitter is missing `vim.api.nvim_cmd` and `vim.api.nvim_exec2`. @smjonas can you also add those into this PR?

Comment (ObserverOfTime) [inline queries/lua/injections.scm:24]: What about `'nvim_exec_lua'` ? :)

Comment (clason) [reply queries/lua/injections.scm:24]: Doesn't really make sense from Lua?

File: queries/lua/injections.scm [hunk 2]
```diff
@@ -8,15 +8,22 @@
...
   arguments: (arguments (string content: _ @query) .))
   (#any-of? @_vimcmd_identifier "vim.treesitter.query.set" "vim.treesitter.query.parse_query" "vim.treesitter.query.parse"))
 
-;; highlight string as query if starts with `;; query`
+; vim.rcprequest(123, "nvim_exec_lua", "return vim.api.nvim_buf_get_lines(0, 0, -1, false)", false)
+((function_call
+  name: (_) @_vimcmd_identifier
+  arguments: (arguments . (_) . (string) @_method . (string content: _ @lua)))
+  (#any-of? @_vimcmd_identifier "vim.rpcrequest" "vim.rpcnotify")
+  (#any-of? @_method "\"nvim_exec_lua\"" "'nvim_exec_lua'" "[[nvim_exec_lua]]"))
```
Comment (ObserverOfTime) [inline queries/lua/injections.scm:24]:
```suggestion
((function_call
  name: (_) @_vimcmd_identifier
  arguments: (arguments . (_) . (string content: _ @_method) . (string content: _ @lua)))
  (#any-of? @_vimcmd_identifier "vim.rpcrequest" "vim.rpcnotify")
  (#eq? @_method "nvim_exec_lua"))
```

**Pull request merged**